### PR TITLE
feat(autoware_adapi_v1_msgs): add velocity factor message

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Add velocity factor message.
https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/ad-api/list/api/planning/velocity_factors/

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by tkhmy — APPROVED** [review]
LGTM

File: autoware_adapi_v1_msgs/planning/msg/VelocityFactor.msg [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+uint16 V2I_GATE_CONTROL_ENTER = 10
+uint16 V2I_GATE_CONTROL_LEAVE = 11
+uint16 MERGE = 12
+uint16 SIDEWALK = 13
+uint16 LANE_CHANGE = 14
+uint16 AVOIDANCE = 15
+uint16 EMERGENCY_STOP_OPERATION = 16
+
+# constants for status
+uint16 APPROACHING = 1
```
[the remark this answers is not in the document]
Comment (isamu-takagi) [reply autoware_adapi_v1_msgs/planning/msg/VelocityFactor.msg:23]: No, it means approaching the stop position regardless of speed. Currently, all states before stopping are this. A deceleration state can be added if it needed.

**Comment by isamu-takagi** [comment]
@yukkysaito @TakaHoribe Is it okay to merge?

File: autoware_adapi_v1_msgs/planning/msg/VelocityFactor.msg (hunk 1 above)
[the remark this answers is not in the document]
Comment (yukkysaito) [reply autoware_adapi_v1_msgs/planning/msg/VelocityFactor.msg:23]: Ok. Thank you!

**Review by yukkysaito — APPROVED** [review]
LGTM

**Review by TakaHoribe — APPROVED** [review]

**Pull request merged**